WordPress/openverse-catalog
# Upgrade to Airflow 2.5.0

## Fixes




Fixes #924 by @AetherUnbound

## Description




> **Warning**
> This PR requires a deployment after merging, as it will break the data refresh & iNaturalist DAGs [^1]

This PR bumps Airflow to 2.5.0. One of the major changes here is the use of the [`common-sql` provider package](https://airflow.apache.org/docs/apache-airflow-providers-common-sql/1.3.0/index.html). The upside of this is that our bespoke `PostgresResultOperator` is completely obsoleted by the `SQLExecuteQueryOperator`! The downside is that this is a **breaking change** which will also require a deployment.

I also moved some of the warning ignorations we had in the `pytest.ini` file into the test that was causing them. This will prevent us from inadvertently ignoring those warnings in other places where we might actually care to know that they're occurring.

Heads up for @rwidom too - this PR affects the iNaturalist DAG which may in-turn cause some problems with #745. I'm happy to help with any merge conflicts that might come up!

[^1]: I tried as hard as I could to avoid this but the addition of the `SQLExecuteQueryOperator` and deprecation of every `[Database]Operator` derivative in the providers packages that come with the 2.5.0 upgrade made trying to perform this upgrade in a non-breaking way nearly impossible. The `PostgresResultOperator` (which is being removed anyway) would require some significant changes that seemed more annoying to figure out than dealing with merging this then immediately deploying.

## Testing Instructions



1. `just recreate && just test --extended`
1. Try running a few ingestion DAGs to make sure they still work
1. If you're set up to do so, run the API stack and try a data refresh to ensure the new operators work

## Checklist



      `Update index.md`).
      a parent feature branch.
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin



<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
